httpgrpc/server/server_test.go

```diff
@@ -73,25 +73,39 @@ func TestBasic(t *testing.T) {
 }
 
 func TestError(t *testing.T) {
-	server, err := newTestServer(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// Does a Fprintln, injecting a newline.
-		http.Error(w, "foo", http.StatusInternalServerError)
-	}))
-	require.NoError(t, err)
-	defer server.grpcServer.GracefulStop()
+	for _, doNotLog := range []bool{true, false} {
+		var stat string
+		if !doNotLog {
+			stat = "not "
+		}
+		t.Run(fmt.Sprintf("test header when DoNotLogErrorHeaderKey is %spresent", stat), func(t *testing.T) {
+			server, err := newTestServer(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if doNotLog {
+					w.Header().Set(DoNotLogErrorHeaderKey, "true")
+				}
+				// Does a Fprintln, injecting a newline.
+				http.Error(w, "foo", http.StatusInternalServerError)
+			}))
+			require.NoError(t, err)
+			defer server.grpcServer.GracefulStop()
 
-	client, err := NewClient(server.URL)
-	require.NoError(t, err)
+			client, err := NewClient(server.URL)
+			require.NoError(t, err)
 
-	req, err := http.NewRequest("GET", "/hello", &bytes.Buffer{})
-	require.NoError(t, err)
+			req, err := http.NewRequest("GET", "/hello", &bytes.Buffer{})
+			require.NoError(t, err)
 
-	req = req.WithContext(user.InjectOrgID(context.Background(), "1"))
-	recorder := httptest.NewRecorder()
-	client.ServeHTTP(recorder, req)
+			req = req.WithContext(user.InjectOrgID(context.Background(), "1"))
+			recorder := httptest.NewRecorder()
+			client.ServeHTTP(recorder, req)
 
-	assert.Equal(t, "foo\n", recorder.Body.String())
-	assert.Equal(t, 500, recorder.Code)
+			assert.Equal(t, "foo\n", recorder.Body.String())
+			assert.Equal(t, 500, recorder.Code)
+			if doNotLog {
+				assert.Equal(t, recorder.Header().Get(DoNotLogErrorHeaderKey), "true")
+			}
```

Comment:
The test should not assert on header, but on whether error produced by Server.Handle is marked for logging or not.

something like:
```golang
func TestServerHandleOptionalError(t *testing.T) {
	for _, doNotLog := range []bool{false, true} {
		h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			if doNotLog {
				w.Header().Set(DoNotLogErrorHeaderKey, "true")
			}
			http.Error(w, "foo", http.StatusInternalServerError)
		})

		s := NewServer(h)
		resp, err := s.Handle(...)
		// check err here
	}
}
```

Reply:
A new test has been added.